### PR TITLE
WFCORE-2836 CachedDcDomainTestCase flakyness

### PR DESCRIPTION
**Pull request by luck3y**
Test for CI & flaky logger add.

### CODE REVIEW

**Comment by luck3y** [comment]
Odd failure here. Not sure if its related, but going to see if it happens again.

**Comment by luck3y** [comment]
retest this please

**Comment by luck3y** [comment]
Odd, I'll open a jira on the hang.

File: testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/cacheddc/CachedDcDomainTestCase.java [hunk 1]
```diff
@@ -166,6 +167,7 @@ public void dcConfigChangePropagated() throws Exception {
         domainManager = DomainTestSupport.create(domainConfig);
         domainManager.start();
 
+        domainManager.getDomainSlaveLifecycleUtil().awaitHostController(System.currentTimeMillis());
```
Comment (bstansberry) [inline testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/cacheddc/CachedDcDomainTestCase.java]: domainManager.start() should be doing this, via its call to DomainLifecycleUtil.start(). Was this not working somehow?

File: testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/cacheddc/CachedDcDomainTestCase.java [hunk 2]
```diff
@@ -194,6 +196,21 @@ public void dcConfigChangePropagatedAfterRestart() throws Exception {
...
+        // occasionally before the slave host unregisters completely, then when the
+        // slave unregisters the op is rolled back.
+        long deadline = System.currentTimeMillis() + 1000;
+        while (true) {
+            if (!domainManager.getDomainSlaveLifecycleUtil().isHostControllerStarted()) {
+                break;
+            }
+            if (System.currentTimeMillis() > deadline) {
+                break;
+            }
```
Comment (bstansberry) [inline testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/cacheddc/CachedDcDomainTestCase.java]: A small sleep in this loop would be good.

File: testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/cacheddc/CachedDcDomainTestCase.java [hunk 3]
```diff
@@ -57,6 +57,7 @@
 
     private static final long TIMEOUT_S = TimeoutUtil.adjust(30);
     private static final long NOT_STARTED_TIMEOUT_S = TimeoutUtil.adjust(7);
+    private static final int TIMEOUT_SLEEP_MILLIS = 300;
```
Comment (bstansberry) [inline testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/cacheddc/CachedDcDomainTestCase.java]: This seems pretty long for a polling sleep. Better than 500 ms, but still pretty long. Maybe 25 or 50? WDYT? I don't think invoking ops that often will add much load to the server, but cutting 250ms off a poll loop a few times will add up.

Comment (luck3y) [inline testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/cacheddc/CachedDcDomainTestCase.java]: There I was thinking cutting it down by 200ms was being adventerous ;) Sure I'll make it 50.

File: testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/cacheddc/CachedDcDomainTestCase.java (hunk 2 above)
Comment (luck3y) [inline testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/cacheddc/CachedDcDomainTestCase.java]: Opps, yes :)

**Comment by luck3y** [comment]
retest this please

**Comment by luck3y** [comment]
retest this please

**Comment by luck3y** [comment]
retest this please

**Pull request merged**